src/k2/cpo/client/Client.h

```diff
@@ -236,6 +249,40 @@ class CPOClient {
         });
     }
 
+    // Check if all partitions are assigned within the deadline. Returns a future of assignment status.
+    template<typename ClockT=Clock>
+    seastar::future<Status> _checkAllParititonsAssigned(Deadline<ClockT> deadline, const String& name, uint8_t retries=5) {
+        if (!retries) {
+            return seastar::make_ready_future<Status>(Statuses::S408_Request_Timeout("check partition assigned retries exceeded"));
+        }
+        return _getAssignedPartitionWithRetry(deadline, name, dto::Key{})
```

Comment:
this is a problem i think for the way the helper _getAssignedPartitionWithRetry works. It uses a cached response if partition 0 has been assigned